### ISSUE TITLE
Resource inheritance

### ISSUE THREAD

**Issue by jvehent** [mozilla/videur#5]
I'm wondering if we need support for resource inheritance, aka defining validation rules and limits on a top-level resource, and have sub-resource directly inherit them.

Take the list of resources below:
1. `/resource`
2. `/resource/something`
3. `/resource/users`
4. `/resource/users/`
5. `/resource/users/<userid>`
6. `/resource/users/<userid>?param=somethingelse`

Some information, like rate limiting, request size or TLS parameters, can easily be shared at the top-level and inherited by the children. Some other options, like query parameters validation, are probably harder to reuse. But a `session` parameter will look the same across all resources, and thus would be interesting to define only once.

Another issue is the one of selecting the most specific path when a query comes in. Should we only do exact matching of request path? Or should we try to find the most specific path and use that for validation? I think the later will be required for resources like `5` and `6`. The inspection algorithm could parse a resource tree, loading and overloading parameters along the way, and using the most-specific resource.

Or is this just too complicated?
